---
title: åŸºäºè›‹ç™½è´¨ç»“æ„è¡¨å¾çš„é¢„è®­ç»ƒæ¨¡å‹
separator: <!--s-->
verticalSeparator: <!--v-->
theme: simple
highlightTheme: github
css: custom.css
revealOptions:
    transition: 'slide'
    transitionSpeed: fast
    center: false
    slideNumber: "c/t"
    width: 1000
---

<div class="middle center">
<div style="width: 100%">
<img src="graph/logo.png" style="margin-bottom: 1em">

# åŸºäºè›‹ç™½è´¨è”åˆè¡¨å¾çš„é¢„è®­ç»ƒæ¨¡å‹SSDM

<hr/>

2024 Spring

By [æ··åˆ2207 æœå®—æ³½](https://github.com/Yaoyaolingbro) [æ··åˆ2206ç­æ¨æ­£å®‡]()

<div style="text-align: right; margin-top: 1em;">
<p>2024.3.15&emsp;&emsp;&emsp;</p>
</div>


</div>
</div>

<!--s-->

<div class="middle center">
<div style="width: 100%">

# Part.1 Introduction

</div>
</div>

<!--v-->
## Protein
è›‹ç™½è´¨ä½œä¸ºç”Ÿå‘½çš„åŸºæœ¬ç»„æˆéƒ¨åˆ†ï¼Œåœ¨ä»é…¶ååº”åˆ°ç»†èƒä¿¡å·ä¼ å¯¼çš„è®¸å¤šç”Ÿç‰©è¿‡ç¨‹ä¸­å‘æŒ¥ç€å…³é”®ä½œç”¨ã€‚

è›‹ç™½è´¨æ˜¯ç”±æ®‹åŸºç»„æˆçš„å¤§åˆ†å­ï¼Œå³æ°¨åŸºé…¸ï¼Œä»¥é“¾çš„å½¢å¼è¿æ¥åœ¨ä¸€èµ·ã€‚å°½ç®¡åªæœ‰20ç§æ ‡å‡†æ®‹åŸºç±»å‹ï¼Œä½†å®ƒä»¬çš„ä¼—å¤šç»„åˆä¿ƒæˆäº†è‡ªç„¶ç•Œä¸­å‘ç°çš„è›‹ç™½è´¨çš„å·¨å¤§å¤šæ ·æ€§ã€‚

è€Œå¯¹è›‹ç™½è´¨çš„æœ‰æ•ˆç†è§£å¯¹äºæ­ç¤ºç–¾ç—…ã€è¯ç‰©å‘ç°å’Œåˆæˆç”Ÿç‰©å­¦çš„æ½œåœ¨æœºåˆ¶è‡³å…³é‡è¦ã€‚

> Focus çš„é‡ç‚¹åœ¨äºå¤æ‚çš„ä¸‰ç»´ç»“æ„å’ŒåŠ¨æ€è¡Œä¸ºï¼Œè¿™æ˜¯å†³å®šå®ƒä»¬çš„åŠŸèƒ½å¤šæ ·æ€§çš„æ ¹æœ¬

<!--v-->

## è›‹ç™½è´¨åŠŸèƒ½
è›‹ç™½è´¨åŠŸèƒ½é¢„æµ‹æ˜¯ç”Ÿç‰©ä¿¡æ¯å­¦ä¸­çš„ä¸€é¡¹é‡è¦ä»»åŠ¡ï¼Œåœ¨**ç–¾ç—…æœºåˆ¶çš„é˜æ˜**å’Œ**è¯ç‰©é¶ç‚¹å‘ç°**ç­‰é¢†åŸŸæœ‰ç€é‡è¦ä½œç”¨ã€‚

**å› ä¸ºä¼ ç»Ÿçš„æµ‹å®šè›‹ç™½è´¨åŠŸèƒ½çš„ç”ŸåŒ–å®éªŒé€šå¸¸æˆæœ¬é«˜ã€è€—æ—¶é•¿ã€é€šé‡ä½ï¼Œæ‰€ä»¥å¼€å‘å‡ºé«˜æ•ˆä¸”å‡†ç¡®çš„è›‹ç™½è´¨åŠŸèƒ½é¢„æµ‹è®¡ç®—æ–¹æ³•ååˆ†é‡è¦**


<!--v-->


## ä»€ä¹ˆæ˜¯Pretrain Modelï¼Ÿ

- Pretrain Modelæ˜¯ä¸€ç§åœ¨å¤§è§„æ¨¡æ•°æ®ä¸Šé¢„è®­ç»ƒçš„æ¨¡å‹(éå¸¸å®½æ³›çš„å®šä¹‰ğŸ˜‚)
- é€šè¿‡é¢„è®­ç»ƒï¼Œæ¨¡å‹å¯ä»¥å­¦ä¹ åˆ°æ•°æ®çš„åˆ†å¸ƒï¼Œæé«˜æ¨¡å‹çš„æ³›åŒ–èƒ½åŠ›
- Pretrain Modelå¯ä»¥ç”¨äºå¤šç§ä»»åŠ¡ï¼Œå¦‚åˆ†ç±»ã€èšç±»ã€ç”Ÿæˆç­‰

> æ‰€è°“çš„Representation Learningï¼Œå…¶å®å°±æ˜¯AIæ¨¡å‹åœ¨åšçš„äº‹æƒ…

> è€ŒPretrain Modelåœ¨å¤§é‡æ•°æ®é©±åŠ¨ä¸‹ä¼šåœ¨æ›´å¤šçš„ä¸‹æ¸¸ä»»åŠ¡ä¸Šè¡¨ç°ä¼˜å¼‚ï¼Œå·²ç»åœ¨ä¼—å¤šè›‹ç™½è´¨ä¸‹æ¸¸ä»»åŠ¡ä¸­å–å¾—äº†æˆåŠŸ

<!--v-->
## ä¸€ç‚¹ç‚¹novelçš„ä¸œè¥¿
Yet, the sheer complexity of protein structures and interactions poses challenges. 

> Figure1: Comparisons of PDB (blue)and AlphaFold (red) structures for GB1 (PDB:2GI9) and SARS-Cov-2 RBD bound to human ACE2 (PDB:6M0J, orange & grey).
![20240529163343.png](graph/20240529163343.png)



<!--v-->
## å…³äºè›‹ç™½è´¨ç»“æ„çš„BackGround
| è›‹ç™½è´¨åŸºç¡€ç»“æ„ | SE(3) diffusion model  |
| --- | --- |
|![20240529162959.png](graph/20240529162959.png)| ![20240529165239.png](graph/20240529165239.png) |

<!--v-->
## è›‹ç™½è´¨çš„é‡è¦æ€§å’ŒæŒ‘æˆ˜ 
- è›‹ç™½è´¨ç”±æ°¨åŸºé…¸çš„çº¿æ€§é“¾ç»„æˆï¼Œå¯æŠ˜å æˆç‰¹å®šçš„æ„è±¡ï¼Œå†³å®šäº†å…¶åŠŸèƒ½
- è›‹ç™½è´¨ç»“æ„ç¡®å®šçš„æŒ‘æˆ˜å¯¼è‡´è›‹ç™½è´¨ç»“æ„æ•°æ®é›†ç›¸å¯¹è¾ƒå°
- æœ€è¿‘çš„å·¥ä½œåˆ©ç”¨æœªæ ‡è®°çš„è›‹ç™½è´¨åºåˆ—æ•°æ®æ¥å­¦ä¹ è›‹ç™½è´¨çš„æœ‰æ•ˆè¡¨ç¤º
- åŸºäºç»“æ„çš„è›‹ç™½è´¨ç¼–ç å™¨å’Œé¢„è®­ç»ƒæ–¹æ³•çš„å‘å±•å¡«è¡¥äº†è›‹ç™½è´¨åŠŸèƒ½æ³¨é‡Šçš„å·®è·
- æœ€æ–°çš„è›‹ç™½è´¨ç»“æ„é¢„æµ‹æ–¹æ³•ä½¿å¾—å¯ä»¥æœ‰æ•ˆåœ°é¢„æµ‹å¤§é‡è›‹ç™½è´¨åºåˆ—çš„ç»“æ„
- ......

> SSF(Structure Sequence Function) is all you need!


<!--s-->

<div class="middle center">
<div style="width: 100%">

# Part.2 Related Work & Method

</div>
</div>

<!--v-->
## Protein Encoder
- å…¨åŸå­çº§åˆ«çš„è›‹ç™½è´¨ç»“æ„ç¼–ç å™¨æ˜¯ååˆ†é‡è¦çš„ï¼ï¼ï¼
> ä¼ ç»Ÿçš„è›‹ç™½è´¨ç»“æ„ç¼–ç å™¨ä¸»è¦åŸºäºå›¾å·ç§¯ç½‘ç»œ(GCN)å’Œè‡ªç¼–ç å™¨(AE) â€”â€” ä½†æ˜¯è¿™äº›æ–¹æ³•åœ¨å¤„ç†å…¨åŸå­çº§åˆ«çš„è›‹ç™½è´¨ç»“æ„æ—¶å­˜åœ¨ä¸€å®šçš„å±€é™æ€§ï¼ˆæˆ‘ä»¬çœŸçš„åº”å½“é‡‡å–è¿™æ ·çš„æ–¹å¼å—ï¼Ÿï¼‰

å› è€Œæˆ‘ä»¬æå‡ºäº†ä¸€ç§æ–°çš„è›‹ç™½è´¨ç»“æ„ç¼–ç å™¨ï¼Œä»¥é€‚ç”¨äºattetion-basedçš„æ¡†æ¶ã€‚

<!--v-->

## Backbone Representation
æˆ‘ä»¬æå‡ºçš„éª¨å¹²çº§è¡¨ç¤ºçš„å›¾ç¤ºå¦‚ä¸‹ï¼š
- ( a )æ°¨åŸºé…¸iéª¨æ¶åæ ‡ç³»çš„æ„å»º
- ( b )è®¡ç®—ä¸¤ä¸ªæ°¨åŸºé…¸iå’Œjçš„éª¨æ¶åæ ‡ç³»ä¹‹é—´çš„3ä¸ªæ¬§æ‹‰è§’
- ( c )è¯´æ˜äº†ç°æœ‰æ–¹æ³•å’Œæˆ‘ä»¬æå‡ºçš„æ–¹æ³•ç¡®å®šæ°¨åŸºé…¸iå’Œjä¹‹é—´çš„ç›¸å¯¹æ—‹è½¬ã€‚ç´«è‰²è™šçº¿è¡¨ç¤ºç°æœ‰æ–¹æ³•å¦‚ä½•é€šè¿‡è®¡ç®—æ²¿é“¾çš„æ‰€æœ‰ä¸»é“¾äºŒé¢è§’æ¥ç¡®å®šiå’Œjä¹‹é—´çš„ç›¸å¯¹æ—‹è½¬ã€‚é»„ç®­å¤´æ˜¾ç¤ºäº†æˆ‘ä»¬çš„æ–¹æ³•å¦‚ä½•ä»…ä½¿ç”¨ä¸‰ä¸ªæ¬§æ‹‰è§’æ¥ç¡®å®šæ°¨åŸºé…¸iå’Œjä¹‹é—´çš„ç›¸å¯¹æ—‹è½¬ã€‚

![20240529214502.png](graph/20240529214502.png)

<!--v-->

## Residua Embedding
æˆ‘ä»¬æå‡ºçš„æ®‹åŸºåµŒå…¥çš„å›¾ç¤ºå¦‚ä¸‹ï¼š
> ä¸alphaFoldçš„æ®‹åŸºåµŒå…¥ç›¸æ¯”ï¼Œæˆ‘ä»¬çš„æ®‹åŸºæœªè€ƒè™‘`180-rotation-symmetric`,ä½†ç”±äºåŸå­åç§°ç»™å®šï¼Œæˆ‘ä»¬æ— éœ€è€ƒè™‘è¿™ä¸ªé—®é¢˜ã€‚
![20240529214817.png](graph/20240529214817.png)


<!--v-->
## Attention-based Model
å…³äºattention-based modelé€‰æ‹©ï¼Œæ˜¯å› ä¸ºå…¶åœ¨å¤„ç†å…¨åŸå­çº§åˆ«çš„è›‹ç™½è´¨ç»“æ„æ—¶å…·æœ‰æ›´å¥½çš„æ€§èƒ½ã€‚
è€Œå…³äºå¦‚ä½•å°†SSï¼ˆStructure & Sequence Joint Informationï¼‰çš„é€‰æ‹©ï¼Œæˆ‘ä»¬åšäº†å¦‚ä¸‹çš„ablation studyï¼š
![20240529220515.png](graph/20240529220515.png)

<!--v-->
## Contrastive Learning
- å¯¹æ¯”å­¦ä¹ æ˜¯ä¸€ç§æ— ç›‘ç£å­¦ä¹ æ–¹æ³•ï¼Œé€šè¿‡æœ€å¤§åŒ–æ­£æ ·æœ¬çš„ç›¸ä¼¼æ€§å’Œè´Ÿæ ·æœ¬çš„å·®å¼‚æ€§æ¥å­¦ä¹ æ•°æ®çš„è¡¨ç¤º

> A blog about Contrastive Learning: [Contrastive Self-Supervised Learning](https://ankeshanand.com/blog/2020/01/26/contrative-self-supervised-learning.html)
> ![20240529213023.png](graph/20240529213023.png)

<!--v-->
## How to Train
ç»“åˆä¸Šè¿°æˆ‘ä»¬æ‰€æä¾›çš„çŸ¥è¯†ï¼Œæˆ‘ä»¬çš„è®­ç»ƒæ¨¡å‹å¦‚ä¸‹:

![20240529220817.png](graph/20240529220817.png)

> Denoise to be continued......


<!--s-->

<div class="middle center">
<div style="width: 100%">

# Part.3 Experiment & Result

</div>
</div>



<!--v-->
## Different prediction
è‡ªé¢„æµ‹æ–¹æ³•ã€‚æˆ‘ä»¬ä½¿ç”¨iã€jã€kã€tæ¥è¡¨ç¤ºé‡‡æ ·çš„æ®‹åŸºç´¢å¼•ã€‚ä»»åŠ¡ä¸å„è‡ªçš„MLPå¤´ç›¸å…³è”ï¼šfresã€fdisã€fangå’Œfdihã€‚CE(Â·)æ˜¯äº¤å‰ç†µæŸå¤±ï¼Œè§’åº¦ç”¨bin(Â·)ç¦»æ•£åŒ–ã€‚

![20240530112247.png](graph/20240530112247.png)

<!-- <section>
\[
\begin{array}{lc}
\hline \text { Method } & \text { Loss function } \\
\hline \text { Residue Type Prediction } & \operatorname{CE}\left(f_{\text {res }}\left(\boldsymbol{z}_{i}\right), r_{i}\right) \\
\text { Distance Prediction } & \left(f_{\text {dis }}\left(\boldsymbol{z}_{i}, \boldsymbol{z}_{j}\right)-\left\|\boldsymbol{x}_{i}-\boldsymbol{x}_{j}\right\|_{2}\right)^{2} \\
\text { Angle Prediction } & \left.\mathrm{CE}\left(f_{\text {ang }} \boldsymbol{z}_{i}, \boldsymbol{z}_{j}, \boldsymbol{z}_{k}\right), \operatorname{bin}(\angle i j k)\right) \\
\text { Dihedral Prediction } & \mathrm{CE}\left(f_{\text {dih }}\left(\boldsymbol{z}_{i}, \boldsymbol{z}_{j}, \boldsymbol{z}_{k}, \boldsymbol{z}_{t}\right), \operatorname{bin}(\angle i j k t)\right) \\
\hline
\end{array}
\]
</section> -->


<!--v-->

## Denoise Details

è¿™äº›æ–¹æ³•å—åˆ°æ‰©æ•£æ¨¡å‹åœ¨æ•æ‰åºåˆ—å’Œç»“æ„çš„è”åˆåˆ†å¸ƒæ–¹é¢çš„æˆåŠŸå¯å‘ã€‚
åœ¨é¢„è®­ç»ƒè¿‡ç¨‹ä¸­ï¼Œå¯¹ç»“æ„å’Œåºåˆ—åº”ç”¨å™ªå£°æ°´å¹³ $t\in\{1,..,T\}$ï¼Œå…¶ä¸­è¾ƒé«˜çš„æ°´å¹³è¡¨ç¤ºè¾ƒå¤§çš„å™ªå£°ã€‚

ç¼–ç å™¨çš„è¡¨ç¤ºç”¨äºé€šè¿‡ä»¥ä¸‹æŸå¤±å‡½æ•°è¿›è¡Œå»å™ªï¼š

<math xmlns="http://www.w3.org/1998/Math/MathML" display="block"><mtable columnspacing="1em" rowspacing="4pt"><mtr><mtd><msub><mrow><mi data-mjx-variant="-tex-calligraphic" mathvariant="script">L</mi></mrow><mrow><mtext>struct&nbsp;</mtext></mrow></msub><mo>=</mo><msub><mrow><mi mathvariant="double-struck">E</mi></mrow><mrow><mi>t</mi><mo>âˆ¼</mo><mo fence="false" stretchy="false">{</mo><mn>1</mn><mo>,</mo><mo>â€¦</mo><mo>,</mo><mi>T</mi><mo fence="false" stretchy="false">}</mo></mrow></msub><msub><mrow><mi mathvariant="double-struck">E</mi></mrow><mrow><mi>Ïµ</mi><mo>âˆ¼</mo><mrow><mi data-mjx-variant="-tex-calligraphic" mathvariant="script">N</mi></mrow><mo stretchy="false">(</mo><mn>0</mn><mo>,</mo><mi>I</mi><mo stretchy="false">)</mo></mrow></msub><mrow data-mjx-texclass="INNER"><mo data-mjx-texclass="OPEN">[</mo><msubsup><mrow data-mjx-texclass="INNER"><mo data-mjx-texclass="OPEN">âˆ¥</mo><mi>Ïµ</mi><mo>âˆ’</mo><msub><mi>f</mi><mrow><mtext>noise&nbsp;</mtext></mrow></msub><mrow data-mjx-texclass="INNER"><mo data-mjx-texclass="OPEN">(</mo><msup><mi mathvariant="bold-italic">z</mi><mrow><mo stretchy="false">(</mo><mi>t</mi><mo stretchy="false">)</mo></mrow></msup><mo>,</mo><msup><mi mathvariant="bold-italic">x</mi><mrow><mo stretchy="false">(</mo><mi>t</mi><mo stretchy="false">)</mo></mrow></msup><mo data-mjx-texclass="CLOSE">)</mo></mrow><mo data-mjx-texclass="CLOSE">âˆ¥</mo></mrow><mrow><mn>2</mn></mrow><mrow><mn>2</mn></mrow></msubsup><mo data-mjx-texclass="CLOSE">]</mo></mrow><mo>,</mo></mtd></mtr><mtr><mtd><msub><mrow><mi data-mjx-variant="-tex-calligraphic" mathvariant="script">L</mi></mrow><mrow><mtext>seq&nbsp;</mtext></mrow></msub><mo>=</mo><msub><mrow><mi mathvariant="double-struck">E</mi></mrow><mrow><mi>t</mi><mo>âˆ¼</mo><mo fence="false" stretchy="false">{</mo><mn>1</mn><mo>,</mo><mo>.</mo><mo>.</mo><mo>,</mo><mi>T</mi><mo fence="false" stretchy="false">}</mo></mrow></msub><munder><mo data-mjx-texclass="OP">âˆ‘</mo><mrow><mi>i</mi></mrow></munder><mi>CE</mi><mo data-mjx-texclass="NONE">â¡</mo><mrow data-mjx-texclass="INNER"><mo data-mjx-texclass="OPEN">(</mo><msub><mi>r</mi><mrow><mi>i</mi></mrow></msub><mo>,</mo><msub><mi>f</mi><mrow><mtext>res&nbsp;</mtext></mrow></msub><mrow data-mjx-texclass="INNER"><mo data-mjx-texclass="OPEN">(</mo><msubsup><mi mathvariant="bold-italic">z</mi><mrow><mi>i</mi></mrow><mrow><mo stretchy="false">(</mo><mi>t</mi><mo stretchy="false">)</mo></mrow></msubsup><mo data-mjx-texclass="CLOSE">)</mo></mrow><mo data-mjx-texclass="CLOSE">)</mo></mrow><mo>,</mo></mtd></mtr></mtable></math>

é€šè¿‡æ‰­æ›²æ‰°åŠ¨ç”Ÿæˆç›¸å…³æ„è±¡ï¼Œå¹¶åœ¨ä¸¤ä¸ªæ‰©æ•£è½¨è¿¹ä¹‹é—´è¿›è¡Œç›¸äº’å»å™ªï¼ŒSiamDiffå¢å¼ºäº†åŸºäºæ‰©æ•£çš„é¢„è®­ç»ƒã€‚

> å…¶ä¸­ï¼Œ$f_{\text{noise}}$å’Œ$f_{\text{res}}$æ˜¯å»å™ªç½‘ç»œã€‚


<!--v-->

## çƒ­ç¨³å®šæ€§

**çƒ­ç¨³å®šæ€§**æ˜¯æŒ‡[ç‰©è´¨](https://zh.wikipedia.org/wiki/%E7%89%A9%E8%B4%A8 "ç‰©è´¨")åœ¨ç›¸å¯¹è¾ƒé«˜çš„[æ¸©åº¦](https://zh.wikipedia.org/wiki/%E6%B8%A9%E5%BA%A6 "æ¸©åº¦")ä¸‹é˜»æ­¢å…¶åŒ–å­¦æˆ–ç‰©ç†ç»“æ„è¢«ç ´åæˆ–å‡ºç°å˜åŒ–çš„èƒ½åŠ›ã€‚å¦‚æœä¸€ä¸ªç‰©è´¨çƒ­ç¨³å®šæ€§å¥½ï¼Œå®ƒé‡åˆ°é«˜æ¸©æ—¶å°±ä¸å®¹æ˜“å‡ºç°[åˆ†è§£ååº”](https://zh.wikipedia.org/wiki/%E5%88%86%E8%A7%A3%E5%8F%8D%E5%BA%94 "åˆ†è§£ååº”")æˆ–ä¸å…¶ä»–ç‰©è´¨[èšåˆ](https://zh.wikipedia.org/wiki/%E8%81%9A%E5%90%88 "èšåˆ")ã€‚[[1]](https://zh.wikipedia.org/zh-cn/%E7%83%AD%E7%A8%B3%E5%AE%9A%E6%80%A7#cite_note-1)è¿™äº›çƒ­ç¨³å®šæ€§å¥½çš„ç‰©è´¨è¢«è§†ä¸ºè€çƒ­ææ–™ï¼Œä»–ä»¬æ˜¯[é˜»ç‡ƒå‰‚](https://zh.wikipedia.org/wiki/%E9%98%BB%E7%87%83%E5%89%82 "é˜»ç‡ƒå‰‚")çš„åŸæ–™ã€‚çƒ­ç¨³å®šæ€§ä¹Ÿæ˜¯ä¸€äº›[è›‹ç™½è´¨](https://zh.wikipedia.org/wiki/%E8%9B%8B%E7%99%BD%E8%B4%A8 "è›‹ç™½è´¨")çš„ç‰¹æ€§ã€‚çƒ­ç¨³å®šæ€§å¥½çš„è›‹ç™½è´¨åœ¨é«˜æ¸©ç¯å¢ƒä¸‹å…¶[è›‹ç™½è´¨ç»“æ„](https://zh.wikipedia.org/wiki/%E8%9B%8B%E7%99%BD%E8%B4%A8%E7%BB%93%E6%9E%84 "è›‹ç™½è´¨ç»“æ„")ä¹Ÿèƒ½ä¿æŒç¨³å®šã€‚

https://www.bopuyun.com/article/94258?zch=seoggs

ç ”ç©¶å‘ç°ï¼ŒæŸäº›å•å…‹éš†æŠ—ä½“è¯ç‰©è™½ç„¶åœ¨ä½“å¤–å®éªŒä¸­è¡¨ç°å‡ºè‰¯å¥½çš„è¯ç‰©æ´»æ€§ï¼Œä½†åœ¨è¿›å…¥ä¸´åºŠè¯•éªŒé˜¶æ®µå´ä¼šé­é‡åˆ°ä½“å†…æ´»æ€§é™ä½çš„é—®é¢˜ï¼»4ï¼½

<!--v-->
## æŠ—ä½“åˆ†å­è®¾è®¡
å› æ­¤åœ¨è¯ç‰©ç ”å‘çš„åˆæœŸå°±è¦å…¼é¡¾å…¶è¯æ•ˆåŠ¨åŠ›å­¦çš„é—®é¢˜ã€‚**æŠ—ä½“è¯ç‰©çš„ç¨³å®šæ€§æ˜¯å½±å“æŠ—ä½“è¯æ•ˆåŠ¨åŠ›å­¦çš„å…³é”®å› ç´ ä¹‹ä¸€**ï¼Œé¦–å…ˆæŠ—ä½“çš„é«˜äº²å’ŒåŠ›ä¸é«˜ç‰¹å¼‚æ€§éƒ½éœ€è¦ä»¥ç¨³å®šçš„ç»“æ„ä¸ºåŸºç¡€ï¼Œè¿™æ˜¯å…¶æ­£ç¡®è¡Œä½¿ç”Ÿç‰©å­¦åŠŸèƒ½çš„ä¿éšœã€‚å…¶æ¬¡ï¼ŒæŠ—ä½“çš„ç¨³å®šæ€§è¶Šé«˜ï¼Œåˆ™å…¶æ–°ç”Ÿè‚½é“¾åœ¨ç»†èƒå†…è£…é…æ—¶äº§ç”Ÿé”™è¯¯æŠ˜å ï¼ˆmis-foldingï¼‰çš„æ¦‚ç‡è¶Šä½ï¼Œå¯æº¶æ€§è¡¨è¾¾é‡ä¹Ÿè¶Šé«˜ï¼»5ï¼½ã€‚

**è‰¯å¥½çš„çƒ­ç¨³å®šæ€§æ‰€å¸¦æ¥çš„ç´§å‡‘ç»“æ„ä½¿æŠ—ä½“çš„è›‹ç™½é…¶åˆ‡ä½ç‚¹æ›´ä¸æ˜“æš´éœ²ï¼Œå¹¶å…¶å½±å“è¯å“ä¿è´¨æœŸåŠå­˜æ”¾æ¡ä»¶ï¼Œå…³ç³»åˆ°è¯ç‰©æˆæœ¬**ã€‚ç›®å‰ï¼Œæé«˜è›‹ç™½è´¨çƒ­ç¨³å®šæ€§çš„æ–¹æ³•ä¸»è¦æœ‰éå…±ä»·ä¿®é¥°ã€åŒ–å­¦ä¿®é¥°ã€æ·»åŠ è›‹ç™½è´¨ç¨³å®šå‰‚ã€è›‹ç™½è´¨å·¥ç¨‹ï¼Œä»¥åŠåœ¨æ¶²ä½“çŠ¶æ€åˆ©ç”¨çŸ¿åŒ–æŠ€æœ¯ç›´æ¥åœ¨è›‹ç™½è¡¨é¢å½¢æˆç£·é…¸é’™çŸ¿åŒ–å¤–å£³ä»¥æé«˜è›‹ç™½è´¨çš„ç¨³å®šæ€§ï¼»6ï¼½ ã€‚

> ç”±æ­¤å¯è§ï¼Œåœ¨ä¿è¯æŠ—ä½“äº²å’ŒåŠ›åŠè¡¨è¾¾é‡ç­‰æ€§è´¨ä¸å—å¤ªå¤§å½±å“çš„æƒ…å†µä¸‹ï¼Œæœ€å¤§ç¨‹åº¦ä¸Šæé«˜å…¶ç¨³å®šæ€§ï¼Œå¯¹æŠ—ä½“è¯ç‰©çš„ç ”å‘å…·æœ‰é‡è¦çš„ç°å®æ„ä¹‰ã€‚

<!--v-->

## Downstream Task

æˆ‘ä»¬åœ¨Atom3Dä¸Šè¯„ä¼°äº†æˆ‘ä»¬æå‡ºçš„æ–¹æ³•åœ¨åŠŸèƒ½æ³¨é‡Šå’Œç»“æ„å±æ€§é¢„æµ‹ä»»åŠ¡ä¸­çš„æœ‰æ•ˆæ€§

Four key downstream tasks are considered:

**1. Enzyme Commission (EC) Number Prediction:** This task involves predicting EC numbers that describe a protein's catalytic behavior in biochemical reactions. It's formulated as 538 binary classification problems based on the third and fourth levels of the EC tree [Webb, 1992]. We use dataset splits from Gligorijevic et al. (2021) and test on sequences with up to 95% identity cutoff.

<!--v-->

**2. Gene Ontology (GO) Term Prediction:** This benchmark includes three tasks: predicting a protein's biological process (BP), molecular function (MF), and cellular component (CC). Each task is framed as multiple binary classification problems based on GO term annotations. We employ dataset splits from Gligorijevic et al. (2021) with a 95% sequence identity cutoff.

**3. Protein Structure Ranking (PSR):** This task involves predicting global distance test scores for structure predictions submitted to the Critical Assessment of Structure Prediction (CASP) [Kryshtafovych et al., 2019]. The dataset is partitioned by competition year.

<!--v-->

**4. Mutation Stability Prediction (MSP):** The goal is to predict if a mutation enhances a protein complex's stability. The dataset is divided based on a 30% sequence identity.

<!--v-->

## Result

![20240530153919.png](graph/20240530153919.png)

<!--v-->

![20240530153940.png](graph/20240530153940.png)

<!-- | **Method**           | **PLM** | **Struct. Info.** | **EC**      |      | **GO-BP**   |      | **GO-MF**   |      | **GO-CC**   |      | **PSR**     |      | **MSP**     | Thermo stability |
| -------------------- | ------- | ----------------- | ----------- | ---- | ----------- | ---- | ----------- | ---- | ----------- | ---- | ----------- | ---- | ----------- | ---------------- |
| **ProtBERT-BFD**     | âœ”ï¸       | âŒ                 | 0.838       |      | 0.279       |      | 0.456       |      | 0.408       |      | -           |      | -           | 0.680            |
| **ESM-2-650M**       | âœ”ï¸       | âŒ                 | _0.880_     |      | _0.460_     |      | _0.661_     |      | _0.445_     |      | -           |      | -           | 0.708            |
| **ESM-SSDM**         | âœ”ï¸       | âœ”ï¸                 | 0.730       |      | 0.356       |      | 0.503       |      | 0.414       |      | 0.708       |      | 0.549       | **0.724**        |
| - w/ serial fusion   | âœ”ï¸       | âœ”ï¸                 | **_0.890_** |      | **_0.488_** |      | **_0.681_** |      | _0.464_     |      | _0.829_     |      | **_0.685_** | -                |
| - w/ parallel fusion | âœ”ï¸       | âœ”ï¸                 | 0.792       |      | 0.384       |      | 0.573       |      | 0.407       |      | 0.760       |      | 0.644       |                  |
| - w/ cross fusion    | âœ”ï¸       | âœ”ï¸                 | 0.884       |      | 0.470       |      | 0.660       |      | 0.462       |      | 0.747       |      | 0.408       |                  |
| **GVP**              | âŒ       | âœ”ï¸                 | 0.489       |      | 0.326       |      | 0.426       |      | 0.420       |      | 0.726       |      | _0.664_     | 0.571            |
| **ESM-GVP**          |         |                   |             |      |             |      |             |      |             |      |             |      |             |                  |
| - w/ serial fusion   | âœ”ï¸       | âœ”ï¸                 | _0.881_     |      | _0.473_     |      | _0.668_     |      | _0.485_     |      | **_0.866_** |      | 0.617       |                  |
| - w/ parallel fusion | âœ”ï¸       | âœ”ï¸                 | 0.872       |      | 0.446       |      | 0.657       |      | 0.455       |      | 0.702       |      | 0.592       |                  |
| - w/ cross fusion    | âœ”ï¸       | âœ”ï¸                 | 0.880       |      | 0.465       |      | 0.664       |      | 0.469       |      | 0.764       |      | 0.583       |                  |
| **CDConv**           | âŒ       | âœ”ï¸                 | 0.820       |      | 0.453       |      | 0.654       |      | **_0.479_** |      | 0.786       |      | 0.529       | 0.660            |
| **ESM-CDConv**       |         |                   |             |      |             |      |             |      |             |      |             |      |             |                  |
| - w/ serial fusion   | âœ”ï¸       | âœ”ï¸                 | _0.880_     |      | _0.465_     |      | 0.658       |      | 0.475       |      | _0.851_     |      | 0.566       |                  |
| - w/ parallel fusion | âœ”ï¸       | âœ”ï¸                 | 0.879       |      | 0.448       |      | _0.662_     |      | 0.455       |      | 0.803       |      | _0.602_     |                  | -->

<!--v-->


## Result
ESM-SSDM (serial fusion) results on EC and GO-MF with different learning rate ratios.
![20240530115734.png](graph/20240530115734.png)

<!--v-->
## Dataset
![20240530160154.png](graph/20240530160154.png)

<!--v-->
## Moreover...
å½“ç„¶ï¼Œæˆ‘ä»¬ä¹Ÿå¯ä»¥å°†ä¸€äº›ä¼—å¤šä¸‹æ¸¸ä»»åŠ¡ä½œä¸ºæˆ‘ä»¬çš„è¯„ä»·æŒ‡æ ‡
![20240530155244.png](graph/20240530155244.png)

<!--s-->

<div class="middle center">
<div style="width: 100%">

# Part.4 Discussion

</div>
</div>

<!--v-->
## å…³äºä¸€ç‚¹ç‚¹RL
æˆ‘å§‹ç»ˆè®¤ä¸ºï¼Œç›®å‰åŸºäºMLPçš„ä¸»æµæ¨¡å‹æ€æƒ³å§‹ç»ˆæºäºä¸€äº›äººç±»æ„è¯†çš„æ¢ç´¢æ€è€ƒï¼Œè€Œå¼ºåŒ–å­¦ä¹ çš„æ½œèƒ½å§‹ç»ˆæœªè¢«å®Œå…¨æŒ–æ˜ï¼Œå°¤å…¶åœ¨for scienceé¢†åŸŸ......
![20240529215428.png](graph/20240529215428.png)

<!--v-->
## å›¾ä¾‹

![20240529220114.png](graph/20240529220114.png)

<!--s-->

<div class="middle center">
<div style="width: 100%">

# æ„Ÿè°¢è†å¬

> æ³è¯·æ‰¹è¯„æŒ‡æ­£ï¼

</div>
</div>

<!--v-->
## å‚è€ƒæ–‡çŒ®
[1]	Bin Ma and Richard Johnson. De novo sequencing and homology searching. Molecular & cellular proteomics, 11(2), 2012.

[2]	Bin Ma. Novor: real-time peptide de novo sequencing software. Journal of the American Society for Mass Spectrometry, 26(11):1885â€“1894, 2015.

[3]	Yue Cao, Payel Das, Vijil Chenthamarakshan, Pin-Yu Chen, Igor Melnyk, and Yang Shen. Fold2seq: A joint sequence(1d)-fold(3d) embedding-based generative model for protein design.

<!--v-->

[4]	Bowen Jing, Stephan Eismann, Pratham N. Soni, and Ron O. Dror. Learning from protein structure with geometric vector perceptrons. In International Conference on Learning Representations, 2021.

[5]	Pedro Hermosilla, Marco SchÃ¤fer, MatË‡ ej Lang, Gloria Fackelmann, Pere Pau VÃ¡zquez, Barbora KozlÃ­kovÃ¡, Michael Krone, Tobias Ritschel, and Timo Ropinski. Intrinsic-extrinsic convolution and pooling for learning on 3d protein structures.

[6]	Ahmed Elnaggar, Michael Heinzinger, Christian Dallago, Ghalia Rehawi, Wang Yu, Llion Jones, Tom Gibbs, Tamas Feher, Christoph Angerer, Martin Steinegger, Debsindhu Bhowmik, and Burkhard Rost. Prottrans: Towards cracking the language of lifes code through self-supervised deep learning and high performance computing.

<!--v-->

[7]	Pedro Hermosilla and Timo Ropinski. Contrastive representation learning for 3d protein structures. In Submitted to The Tenth International Conference on Learning Representations, 2022.

[8]	Limei Wang, Haoran Liu, Yi Liu, Jerry Kurtin, and Shuiwang Ji. Learning protein representations via complete 3d graph networks. ArXiv, abs/2207.12600, 2022a.

[9]	John Jumper, Richard Evans, Alexander Pritzel, Tim Green, Michael Figurnov, Olaf Ronneberger, Kathryn Tunyasuvunakool, Russ Bates, Augustin Å½Ã­dek, Anna Potapenko, et al. Highly accurate protein structure prediction with alphafold. Nature, 596(7873):583â€“589, 2021.
