---
title: åŸºäºè›‹ç™½è´¨ç»“æ„è¡¨å¾çš„é¢„è®­ç»ƒæ¨¡å‹
separator: <!--s-->
verticalSeparator: <!--v-->
theme: simple
highlightTheme: github
css: custom.css
revealOptions:
    transition: 'slide'
    transitionSpeed: fast
    center: false
    slideNumber: "c/t"
    width: 1000
---

<div class="middle center">
<div style="width: 100%">
<img src="graph/logo.png" style="margin-bottom: 1em">

# åŸºäºè›‹ç™½è´¨ç»“æ„è¡¨å¾çš„é¢„è®­ç»ƒæ¨¡å‹

<hr/>

2024 Spring

By [æ··åˆ2207 æœå®—æ³½](https://github.com/Yaoyaolingbro)

<div style="text-align: right; margin-top: 1em;">
<p>2024.3.15&emsp;&emsp;&emsp;</p>
</div>


</div>
</div>

<!--s-->

<div class="middle center">
<div style="width: 100%">

# Part.1 Introduction

</div>
</div>

<!--v-->
## Protein
è›‹ç™½è´¨ä½œä¸ºç”Ÿå‘½çš„åŸºæœ¬ç»„æˆéƒ¨åˆ†ï¼Œåœ¨ä»é…¶ååº”åˆ°ç»†èƒä¿¡å·ä¼ å¯¼çš„è®¸å¤šç”Ÿç‰©è¿‡ç¨‹ä¸­å‘æŒ¥ç€å…³é”®ä½œç”¨ã€‚

è›‹ç™½è´¨æ˜¯ç”±æ®‹åŸºç»„æˆçš„å¤§åˆ†å­ï¼Œå³æ°¨åŸºé…¸ï¼Œä»¥é“¾çš„å½¢å¼è¿æ¥åœ¨ä¸€èµ·ã€‚å°½ç®¡åªæœ‰20ç§æ ‡å‡†æ®‹åŸºç±»å‹ï¼Œä½†å®ƒä»¬çš„ä¼—å¤šç»„åˆä¿ƒæˆäº†è‡ªç„¶ç•Œä¸­å‘ç°çš„è›‹ç™½è´¨çš„å·¨å¤§å¤šæ ·æ€§ã€‚

è€Œå¯¹è›‹ç™½è´¨çš„æœ‰æ•ˆç†è§£å¯¹äºæ­ç¤ºç–¾ç—…ã€è¯ç‰©å‘ç°å’Œåˆæˆç”Ÿç‰©å­¦çš„æ½œåœ¨æœºåˆ¶è‡³å…³é‡è¦ã€‚

> Focus çš„é‡ç‚¹åœ¨äºå¤æ‚çš„ä¸‰ç»´ç»“æ„å’ŒåŠ¨æ€è¡Œä¸ºï¼Œè¿™æ˜¯å†³å®šå®ƒä»¬çš„åŠŸèƒ½å¤šæ ·æ€§çš„æ ¹æœ¬

<!--v-->


## ä»€ä¹ˆæ˜¯Pretrain Modelï¼Ÿ

- Pretrain Modelæ˜¯ä¸€ç§åœ¨å¤§è§„æ¨¡æ•°æ®ä¸Šé¢„è®­ç»ƒçš„æ¨¡å‹(éå¸¸å®½æ³›çš„å®šä¹‰ğŸ˜‚)
- é€šè¿‡é¢„è®­ç»ƒï¼Œæ¨¡å‹å¯ä»¥å­¦ä¹ åˆ°æ•°æ®çš„åˆ†å¸ƒï¼Œæé«˜æ¨¡å‹çš„æ³›åŒ–èƒ½åŠ›
- Pretrain Modelå¯ä»¥ç”¨äºå¤šç§ä»»åŠ¡ï¼Œå¦‚åˆ†ç±»ã€èšç±»ã€ç”Ÿæˆç­‰

> æ‰€è°“çš„Representation Learningï¼Œå…¶å®å°±æ˜¯AIæ¨¡å‹åœ¨åšçš„äº‹æƒ…

> è€ŒPretrain Modelåœ¨å¤§é‡æ•°æ®é©±åŠ¨ä¸‹ä¼šåœ¨æ›´å¤šçš„ä¸‹æ¸¸ä»»åŠ¡ä¸Šè¡¨ç°ä¼˜å¼‚ï¼Œå·²ç»åœ¨ä¼—å¤šè›‹ç™½è´¨ä¸‹æ¸¸ä»»åŠ¡ä¸­å–å¾—äº†æˆåŠŸ

<!--v-->
## ä¸€ç‚¹ç‚¹novelçš„ä¸œè¥¿
Yet, the sheer complexity of protein structures and interactions poses challenges. 

> Figure1: Comparisons of PDB (blue)and AlphaFold (red) structures for GB1 (PDB:2GI9) and SARS-Cov-2 RBD bound to human ACE2 (PDB:6M0J, orange & grey).
![20240529163343.png](graph/20240529163343.png)



<!--v-->
## å…³äºè›‹ç™½è´¨ç»“æ„çš„BackGround
| è›‹ç™½è´¨åŸºç¡€ç»“æ„ | SE(3) diffusion model  |
| --- | --- |
|![20240529162959.png](graph/20240529162959.png)| ![20240529165239.png](graph/20240529165239.png) |

<!--v-->
## è›‹ç™½è´¨çš„é‡è¦æ€§å’ŒæŒ‘æˆ˜ 
- è›‹ç™½è´¨ç”±æ°¨åŸºé…¸çš„çº¿æ€§é“¾ç»„æˆï¼Œå¯æŠ˜å æˆç‰¹å®šçš„æ„è±¡ï¼Œå†³å®šäº†å…¶åŠŸèƒ½
- è›‹ç™½è´¨ç»“æ„ç¡®å®šçš„æŒ‘æˆ˜å¯¼è‡´è›‹ç™½è´¨ç»“æ„æ•°æ®é›†ç›¸å¯¹è¾ƒå°
- æœ€è¿‘çš„å·¥ä½œåˆ©ç”¨æœªæ ‡è®°çš„è›‹ç™½è´¨åºåˆ—æ•°æ®æ¥å­¦ä¹ è›‹ç™½è´¨çš„æœ‰æ•ˆè¡¨ç¤º
- åŸºäºç»“æ„çš„è›‹ç™½è´¨ç¼–ç å™¨å’Œé¢„è®­ç»ƒæ–¹æ³•çš„å‘å±•å¡«è¡¥äº†è›‹ç™½è´¨åŠŸèƒ½æ³¨é‡Šçš„å·®è·
- æœ€æ–°çš„è›‹ç™½è´¨ç»“æ„é¢„æµ‹æ–¹æ³•ä½¿å¾—å¯ä»¥æœ‰æ•ˆåœ°é¢„æµ‹å¤§é‡è›‹ç™½è´¨åºåˆ—çš„ç»“æ„
- ......

> SSF(Structure Sequence Function) is all you need!


<!--s-->

<div class="middle center">
<div style="width: 100%">

# Part.2 Related Work & Method

</div>
</div>

<!--v-->
## Protein Encoder
- å…¨åŸå­çº§åˆ«çš„è›‹ç™½è´¨ç»“æ„ç¼–ç å™¨æ˜¯ååˆ†é‡è¦çš„ï¼ï¼ï¼
> ä¼ ç»Ÿçš„è›‹ç™½è´¨ç»“æ„ç¼–ç å™¨ä¸»è¦åŸºäºå›¾å·ç§¯ç½‘ç»œ(GCN)å’Œè‡ªç¼–ç å™¨(AE) â€”â€” ä½†æ˜¯è¿™äº›æ–¹æ³•åœ¨å¤„ç†å…¨åŸå­çº§åˆ«çš„è›‹ç™½è´¨ç»“æ„æ—¶å­˜åœ¨ä¸€å®šçš„å±€é™æ€§ï¼ˆæˆ‘ä»¬çœŸçš„åº”å½“é‡‡å–è¿™æ ·çš„æ–¹å¼å—ï¼Ÿï¼‰

å› è€Œæˆ‘ä»¬æå‡ºäº†ä¸€ç§æ–°çš„è›‹ç™½è´¨ç»“æ„ç¼–ç å™¨ï¼Œä»¥é€‚ç”¨äºattetion-basedçš„æ¡†æ¶ã€‚

<!--v-->

## Backbone Representation
æˆ‘ä»¬æå‡ºçš„éª¨å¹²çº§è¡¨ç¤ºçš„å›¾ç¤ºå¦‚ä¸‹ï¼š
- ( a )æ°¨åŸºé…¸iéª¨æ¶åæ ‡ç³»çš„æ„å»º
- ( b )è®¡ç®—ä¸¤ä¸ªæ°¨åŸºé…¸iå’Œjçš„éª¨æ¶åæ ‡ç³»ä¹‹é—´çš„3ä¸ªæ¬§æ‹‰è§’
- ( c )è¯´æ˜äº†ç°æœ‰æ–¹æ³•å’Œæˆ‘ä»¬æå‡ºçš„æ–¹æ³•ç¡®å®šæ°¨åŸºé…¸iå’Œjä¹‹é—´çš„ç›¸å¯¹æ—‹è½¬ã€‚ç´«è‰²è™šçº¿è¡¨ç¤ºç°æœ‰æ–¹æ³•å¦‚ä½•é€šè¿‡è®¡ç®—æ²¿é“¾çš„æ‰€æœ‰ä¸»é“¾äºŒé¢è§’æ¥ç¡®å®šiå’Œjä¹‹é—´çš„ç›¸å¯¹æ—‹è½¬ã€‚é»„ç®­å¤´æ˜¾ç¤ºäº†æˆ‘ä»¬çš„æ–¹æ³•å¦‚ä½•ä»…ä½¿ç”¨ä¸‰ä¸ªæ¬§æ‹‰è§’æ¥ç¡®å®šæ°¨åŸºé…¸iå’Œjä¹‹é—´çš„ç›¸å¯¹æ—‹è½¬ã€‚

![20240529214502.png](graph/20240529214502.png)

<!--v-->

## Residua Embedding
æˆ‘ä»¬æå‡ºçš„æ®‹åŸºåµŒå…¥çš„å›¾ç¤ºå¦‚ä¸‹ï¼š
> ä¸alphaFoldçš„æ®‹åŸºåµŒå…¥ç›¸æ¯”ï¼Œæˆ‘ä»¬çš„æ®‹åŸºæœªè€ƒè™‘`180-rotation-symmetric`,ä½†ç”±äºåŸå­åç§°ç»™å®šï¼Œæˆ‘ä»¬æ— éœ€è€ƒè™‘è¿™ä¸ªé—®é¢˜ã€‚
![20240529214817.png](graph/20240529214817.png)


<!--v-->
## Attention-based Model
å…³äºattention-based modelé€‰æ‹©ï¼Œæ˜¯å› ä¸ºå…¶åœ¨å¤„ç†å…¨åŸå­çº§åˆ«çš„è›‹ç™½è´¨ç»“æ„æ—¶å…·æœ‰æ›´å¥½çš„æ€§èƒ½ã€‚
è€Œå…³äºå¦‚ä½•å°†SSï¼ˆStructure & Sequence informationï¼‰çš„é€‰æ‹©ï¼Œæˆ‘ä»¬åšäº†å¦‚ä¸‹çš„ablation studyï¼š
![20240529220515.png](graph/20240529220515.png)

<!--v-->
## Contrastive Learning
- å¯¹æ¯”å­¦ä¹ æ˜¯ä¸€ç§æ— ç›‘ç£å­¦ä¹ æ–¹æ³•ï¼Œé€šè¿‡æœ€å¤§åŒ–æ­£æ ·æœ¬çš„ç›¸ä¼¼æ€§å’Œè´Ÿæ ·æœ¬çš„å·®å¼‚æ€§æ¥å­¦ä¹ æ•°æ®çš„è¡¨ç¤º

> A blog about Contrastive Learning: [Contrastive Self-Supervised Learning](https://ankeshanand.com/blog/2020/01/26/contrative-self-supervised-learning.html)
> ![20240529213023.png](graph/20240529213023.png)

<!--v-->
## How to Train
ç»“åˆä¸Šè¿°æˆ‘ä»¬æ‰€æä¾›çš„çŸ¥è¯†ï¼Œæˆ‘ä»¬çš„è®­ç»ƒæ¨¡å‹å¦‚ä¸‹:

![20240529220817.png](graph/20240529220817.png)

> Denoise to be continued......


<!--s-->

<div class="middle center">
<div style="width: 100%">

# Part.3 Experiment & Result

</div>
</div>

<!--s-->

<div class="middle center">
<div style="width: 100%">

# Part.4 Discussion

</div>
</div>

<!--v-->
## å…³äºä¸€ç‚¹ç‚¹RL
æˆ‘å§‹ç»ˆè®¤ä¸ºï¼Œç›®å‰åŸºäºMLPçš„ä¸»æµæ¨¡å‹æ€æƒ³å§‹ç»ˆæºäºä¸€äº›äººç±»æ„è¯†çš„æ¢ç´¢æ€è€ƒï¼Œè€Œå¼ºåŒ–å­¦ä¹ çš„æ½œèƒ½å§‹ç»ˆæœªè¢«å®Œå…¨æŒ–æ˜ï¼Œå°¤å…¶åœ¨for scienceé¢†åŸŸ......
![20240529215428.png](graph/20240529215428.png)

<!--v-->
## å›¾ä¾‹

![20240529220114.png](graph/20240529220114.png)

<!--s-->

<div class="middle center">
<div style="width: 100%">

# æ„Ÿè°¢è†å¬

> æ³è¯·æ‰¹è¯„æŒ‡æ­£ï¼

</div>
</div>

<!--v-->
## å‚è€ƒæ–‡çŒ®
[1]	Bin Ma and Richard Johnson. De novo sequencing and homology searching. Molecular & cellular proteomics, 11(2), 2012.

[2]	Bin Ma. Novor: real-time peptide de novo sequencing software. Journal of the American Society for Mass Spectrometry, 26(11):1885â€“1894, 2015.

[3]	Yue Cao, Payel Das, Vijil Chenthamarakshan, Pin-Yu Chen, Igor Melnyk, and Yang Shen. Fold2seq: A joint sequence(1d)-fold(3d) embedding-based generative model for protein design.

<!--v-->

[4]	Bowen Jing, Stephan Eismann, Pratham N. Soni, and Ron O. Dror. Learning from protein structure with geometric vector perceptrons. In International Conference on Learning Representations, 2021.

[5]	Pedro Hermosilla, Marco SchÃ¤fer, MatË‡ ej Lang, Gloria Fackelmann, Pere Pau VÃ¡zquez, Barbora KozlÃ­kovÃ¡, Michael Krone, Tobias Ritschel, and Timo Ropinski. Intrinsic-extrinsic convolution and pooling for learning on 3d protein structures.

[6]	Ahmed Elnaggar, Michael Heinzinger, Christian Dallago, Ghalia Rehawi, Wang Yu, Llion Jones, Tom Gibbs, Tamas Feher, Christoph Angerer, Martin Steinegger, Debsindhu Bhowmik, and Burkhard Rost. Prottrans: Towards cracking the language of lifes code through self-supervised deep learning and high performance computing.

<!--v-->

[7]	Pedro Hermosilla and Timo Ropinski. Contrastive representation learning for 3d protein structures. In Submitted to The Tenth International Conference on Learning Representations, 2022.

[8]	Limei Wang, Haoran Liu, Yi Liu, Jerry Kurtin, and Shuiwang Ji. Learning protein representations via complete 3d graph networks. ArXiv, abs/2207.12600, 2022a.

[9]	John Jumper, Richard Evans, Alexander Pritzel, Tim Green, Michael Figurnov, Olaf Ronneberger, Kathryn Tunyasuvunakool, Russ Bates, Augustin Å½Ã­dek, Anna Potapenko, et al. Highly accurate protein structure prediction with alphafold. Nature, 596(7873):583â€“589, 2021.
